rfcs/0098-community-team.md

```diff
@@ -0,0 +1,190 @@
+
+The team should, whenever possible, lead by example. This may include mediating disputes, either privately or publicly as appropriate. It may also include asking community members to refrain from certain topics of discussion or from certain conflicts. Community members are expected to follow these directions, or to appeal them privately by escalating to the full community team. It is valid and important for community members to give feedback on the community team's actions, but team members' decisions are not to be mocked, flaunted, or relitigated in the court of public opinion.
+
+The above are some suggested remedies, but these should not be taken as prescriptive. The full space of social actions is open to team members as potential tools for resolving disputes. Team members are expected to use their authority only in ways that embody justice and that are consistent with the community's values, as formalized in the code of conduct and statement of values. Team members are expected to recognize the power dynamic that they are part of, and to be proactive and explicit in making sure community members know when they're speaking in an official capacity and when they're speaking in a personal capacity. Team members are not expected to be impartial in the sense of pretending not to have personal beliefs or goals, but are expected to hold the community's values above their own, in their official actions.
+
+The community team will hold its members accountable to these ideals.
+
+## Code of Conduct and Statement of Values
+[values]: #values
+
+Following the adoption of this RFC, the code of conduct and statement of values will be displayed on a dedicated page on the NixOS website. They will read as follows, unless amended by a future RFC.
+
+The Nix community recognizes that its members come from many walks of life, and that it is natural for us to disagree on some things. Nonetheless, we are all committed to certain values, to allow us to work together and build awesome technologies.
+
+Community members are expected to follow these values in their interactions in all official Nix social spaces, including websites, chat rooms, bug threads, and conferences. We have a community team who will set and enforce social norms, to make that easier for everyone. faCommunity members are expected to follow the community team's requests.
+
+If you have opinions on the community team's work, get involved! It's a rotating team, and we want to develop the leadership potential in all community members.
+
+Our values are as follows.
+
+**Collaboration** — We all want to make the best technologies we can. We work together on our common goals, with open discussion, and without petty infighting or focusing on blame. We are all, always, teaching each other to be better.
```

Comment:
Wouldn't it be better to acknowledge that first we work on figuring out which of our goals are actually common, so that we can work on them together?

Reply:
The goal isn't to tell people to collaborate if they don't, in fact, have common goals. It's just to talk about what happens when people discover that they do share those goals. This being a technical community, I imagine the vast majority of goals will be technical in nature, so of course it will depend on what everyone is working on.

Reply:
A lot of work done in the community is about distilling which parts of the goals are common (and how to make non-common goal at least non-conflicting — at least if feasible). As written, the sentence is too easy to misread as trivialising this work, or as claiming that we mostly already do have common goals w.r.t. to Nix*

Reply:
Hmm, yeah. I need to chew on that, it's quite fair.

Reply:
Perhaps:
```
**Collaboration** — As we discover common goals with open discussion we work together to make the best technologies we can.
```

Reply:
@7c6f434c Hmm. Arguably "best you can" does reference perfectionism. I had hoped that talking about making good technology would be something that unites people! I wonder if there's a way to capture that which doesn't have the problem - do you have any suggestions?

Reply:
> Perhaps:
> 
> ```
> **Collaboration** — As we discover common goals with open discussion we work together to make the best technologies we can.
> ```

So, as I referenced elsewhere, we're not trying to simplify this for the sake of simplifying it. I would therefore not want to shorten it to just this sentence, because the other sentences are also important substance. The closest thing to your proposal that I'd want to try is something like:

> **Collaboration** -- We all want to make the best technologies we can. As we discover our common goals we work together on them, with open discussion, and without petty infighting or focusing on blame. We are all, always, teaching each other to be better.

I appreciate your efforts to condense things but I need to stress that these values need to stay grounded in reality, and saying each piece of them separately is a benefit to that. The community team has an obligation to relate enforcement actions to these values, as discussed in the "rules vs. principles" section. Overly condensed wording makes that more difficult.

https://github.com/IreneKnapp/rfcs/blob/community/rfcs/0098-community-team.md#rules-vs-principles

Reply:
(also the new wording does not seem to avoid trivialising the goal-alignment work)

Reply:
```
Collaboration: to share the work of building tools, we work to find out which of our goals we have in common, then work together to make our tools better for most of us.
```

(I think in any case it is better to agree of the headline before nitpicking the rest of the paragrapth)

Reply:
> (also the new wording does not seem to avoid trivialising the goal-alignment work)

Yeah, sorry, there are two proposed changes here, yours and @tomberek's and I'm trying to engage with each of them. It's a balancing act, I really appreciate how much everyone is contributing but it's a lot to keep up with. Maybe we should instead be trying to synthesize them both together?

Reply:
@IreneKnapp I do like your latest iteration, but I'd still want to chop off the end.
```
Collaboration -- We all want to make the best technologies we can. As we discover our common goals we work together on them with open discussion.
```

For @7c6f434c 's suggestion. I like it, it references that what we do involves shared work in addition to common goals. Slight changes to wording and more ruthless chopping and trying to retain the "open discussion" part:
```
Collaboration: to share the effort of building better tools, we discover our common goals through open discussion and working together
```

Reply:
I feel like we do need the negative elements, the condemnation of "petty infighting or focusing on blame". Listing the positive elements is more important, since those act as a beacon for good behavior, but good behavior means nothing if there isn't some notion of what it would mean to fail at it.

I do think the last sentence ("teaching each other to be better") is flowery but I also can think of situations where I'd want to appeal to it. Remember, part of the function of these values is to help new members of the community team to understand what their job is. There's no better way to do that than to make it part of the job (to "uphold the community values").

Reply:
> but good behavior means nothing if there isn't some notion of what it would mean to fail at it.

I think it is really important to leave the judgment of failure to the judging body so to make this an engaging document that people are actually willing to buy in.

I, for example.don't relate at all with the word "infighting", so its presence disqualifies the relevance of the broader idea. My brain can know better, but still doesn't trump my feelings while reading.

Reply:
Hi @blaggacao! Glad to see you weighing in here.

The thing is that if you don't have something concrete to anchor it, the idea gets whittled down over time.

Reply:
> Remember, part of the function of these values is to help new members of the community team to understand what their job is. There's no better way to do that than to make it part of the job (to "uphold the community values").

New members should be expected to behave in a way respectful of community values. Putting such an expectation of upholding whatever does definitely imply that people like me are not welcome. I definitely value collaboration that goes well despite acknowledging being somewhat alien to the community, not fully sharing the values just respecting the communucation norms, etc. I think the approach you claim here destroys a valuable trait of Nix* community.

Reply:
> The thing is that if you don't have something concrete to anchor it, the idea gets whittled down over time.

Well, this majorily assumes a dull enactment body. But with a full enactment body, the whole RFC is rendered pointless.

If there is dynamic, it can also get "whittled up". It's the dynamic and fair process boundaries that we should most care about.

Not specific outcome boundaries.

Process justice vs outcome justice.

Reply:
Both of you have good thoughts here and I want to acknowledge the feedback.

I do think that "uphold" doesn't mean the same as "go around promoting", at least for ordinary members of the community, and I hope that that helps a bit, @7c6f434c. It is ultimately asking for behavior, not for belief.

I'm sure you've already seen given how much back and forth we've had, but there are several places in the RFC where we did talk about accepting people's differences.

> The Nix community recognizes that its members come from many walks of life, and that it is natural for us to disagree on some things. Nonetheless, we are all committed to certain values, to allow us to work together and build awesome technologies.

> We strive to be fair and equitable, and to respect the ways people are different.

Both these lines from [the statement of values](https://github.com/IreneKnapp/rfcs/blob/community/rfcs/0098-community-team.md#code-of-conduct-and-statement-of-values), in my view, absolutely include you in the people who benefit from them. If there's something we could do to make it more explicit for your benefit, please let us know.

@blaggacao Yeah. I'm still thinking about this, and about the private discussion we had on positive vs. negative ideals. I think you make good points and I don't yet know where I stand on them. We'll continue to chew on it.

Reply:
> I do think that "uphold" doesn't mean the same as "go around promoting", at least for ordinary members of the community, and I hope that that helps a bit, @7c6f434c. It is ultimately asking for behavior, not for belief.

I think «uphold» is noticeably stronger than «act in conformance with», and yes I do think that «minimality» is an actual good value of this project that your document risks damaging.

Basically, in a behaviour-first way, I find it suspicious when you request me to be committed to any values at all. Because this sounds non-localised.